src/CombatManager.cpp

```diff
@@ -83,7 +83,49 @@ void CombatManager::pcAttackNpcs(CNSocket *sock, CNPacketData *data) {
     }
 }
 
-void CombatManager::combatBegin(CNSocket *sock, CNPacketData *data) {} // stub
+void CombatManager::combatBegin(CNSocket *sock, CNPacketData *data) {
+    sP_CL2FE_REQ_PC_COMBAT_BEGIN* pkt = (sP_CL2FE_REQ_PC_COMBAT_BEGIN*)data->buf;
+    Player* plr = PlayerManager::getPlayer(sock);
+
+    int32_t* pktdata = (int32_t*)((uint8_t*)data->buf + sizeof(sP_CL2FE_REQ_PC_COMBAT_BEGIN));
+
+    size_t resplen = sizeof(sP_FE2CL_NPC_ATTACK_PCs) + sizeof(sAttackResult);
+    uint8_t respbuf[4096];
+
+    memset(respbuf, 0, resplen);
+
+    sP_FE2CL_NPC_ATTACK_PCs* resp = (sP_FE2CL_NPC_ATTACK_PCs*)respbuf;
+    sAttackResult* respdata = (sAttackResult*)(respbuf + sizeof(sP_FE2CL_NPC_ATTACK_PCs));
+
+    pkt->iPC_ID = plr->iID;
```

Comment:
This line is assigning the player's ID to the request packet, not the response. This does nothing.